### ISSUE TITLE
Question: Multiple speakers voice for training data

### ISSUE THREAD

**Issue by rild** [keithito/tacotron#72]
## TL;DR
Is it better to use the same speaker's voice data for learning?

##  Question
I am a student studying speech synthesis at university in Japan.

I read paper about [Tacotron](https://arxiv.org/abs/1703.10135) and wanted to try it myself.
So, I try to train tacotron in Japanese.
(I was able to confirm that training in English works as intended when using the training data available in the repository)

The problem is that encoder/decoder alignment for Japanese would not be learned well.

Here is `step-23000-align.png`
<img width=400 src="https://gyazo.com/0557db9da09a5ce3f9d1a73828640cf7.png"/>

For training, the data used is speech data from three speakers, each reading 100 sentences.

I suppose the reason why alignment fails, is as follows:
- Too few training data samples
- It is learning data of 3 different speakers (multiple speakers)

Thank you.

**Comment by AzamRabiee** [comment]
I usually work with a toy dataset almost similar size as yours for sanity check; but the alignment shows up quickly.
try two things: 1. injection of speaker id as issue #18 . and 2. Monotonic attention

**Comment by rild** [comment]
@AzamRabiee , Thank you for your reaction!

I didn't know the paper: [Monotonic attention](https://arxiv.org/abs/1705.08091)

I would like to read [the issue]( #18 ) and monotonic paper, and try that!

**Comment by rild** [comment]
#53

**Comment by r9y9** [comment]
If you are building multi-speaker models (like DeepVoice2, or 3), it should be okay to use data from multiple speakers. However, I'm pretty sure that the reason you got the non-monotonic alignment is that you don't have sufficient data. https://sites.google.com/site/shinnosuketakamichi/publication/jsut is a freely available Japanese dataset that might be useful for you. The dataset consists of 10 hours audio recordings of a single female speaker. I just started to explore the dataset today with DeepVoice3 architecture and can get nearly monotonic attention very quickly as follows:

![step000005000_layer_1_alignment](https://user-images.githubusercontent.com/1220272/32555041-fc75728e-c4de-11e7-981d-084b23666209.png)

Code is available at https://github.com/r9y9/deepvoice3_pytorch. If you are interested, feel free to contact me.

**Comment by rild** [comment]
@r9y9 , Thank you!

I was just trying to make simplest TTS system, so there was no intention to build multi-speaker model now.

I got the training dataset from [here](http://voice-statistics.github.io/).

I feel the corpuses seems to be too long to train tacotron.
Yesterday, I run `train.py` with 100 speech data (single, no-emotional speech dataset), but I didn't get good result.
 
[the dataset (arxiv)](https://arxiv.org/abs/1711.00354) looks good for training!

**Comment by rild** [comment]
@r9y9
>  If you are interested, feel free to contact me.

I feel very encouraged to hear that.
Thank you so much.
I would like to send a email you later.
